nagios/stackstate_checks/nagios/nagios.py

```diff
@@ -0,0 +1,442 @@
+    # ACKNOWLEDGE_SVC_PROBLEM;<host_name>;<service_description>;<sticky>;<notify>;<persistent>;<author>;<comment>
+    # [1305832665] EXTERNAL COMMAND: ACKNOWLEDGE_SVC_PROBLEM;ip-10-202-161-236;Resources ETL;2;1;0;datadog;alq checking
+    'ACKNOWLEDGE_SVC_PROBLEM': namedtuple(
+        'E_ServiceAck',
+        'host, check_name, sticky_ack, notify_ack, persistent_ack, ack_author, payload'
+    ),
+
+    # Command Format:
+    # ACKNOWLEDGE_HOST_PROBLEM;<host_name>;<sticky>;<notify>;<persistent>;<author>;<comment>
+    'ACKNOWLEDGE_HOST_PROBLEM': namedtuple('E_HostAck',
+                                           'host, sticky_ack, notify_ack, persistent_ack, ack_author, payload'),
+
+    # Comment Format:
+    # PROCESS_SERVICE_CHECK_RESULT;<host_name>;<service_description>;<result_code>;<comment>
+    # We ignore it because Nagios will log a "PASSIVE SERVICE CHECK" after
+    # receiving this, and we don't want duplicate events to be counted.
+    'PROCESS_SERVICE_CHECK_RESULT': False,
+
+    # Host Downtime
+    # [1297894825] HOST DOWNTIME ALERT: ip-10-114-89-59;STARTED; Host has entered a period of scheduled downtime
+    # [1297894825] SERVICE DOWNTIME ALERT: ip-10-114-237-165;intake;STARTED; Service has entered a period of scheduled downtime # noqa: E501
+
+    'HOST DOWNTIME ALERT': namedtuple('E_HostDowntime', 'host, downtime_start_stop, payload'),
+    'SERVICE DOWNTIME ALERT': namedtuple('E_ServiceDowntime', 'host, check_name, downtime_start_stop, payload'),
+}
+
+# Regex for the Nagios event log
+RE_LINE_REG = re.compile(r'^\[(\d+)\] EXTERNAL COMMAND: (\w+);(.*)$')
+RE_LINE_EXT = re.compile(r'^\[(\d+)\] ([^:]+): (.*)$')
+
+
+class NagiosCheck(AgentCheck):
+    SERVICE_CHECK_NAME = INSTANCE_TYPE = "nagios"
+    NAGIOS_CONF_KEYS = [
+        re.compile(r'^(?P<key>log_file)\s*=\s*(?P<value>.+)$'),
+        re.compile(r'^(?P<key>host_perfdata_file_template)\s*=\s*(?P<value>.+)$'),
+        re.compile(r'^(?P<key>service_perfdata_file_template)\s*=\s*(?P<value>.+)$'),
+        re.compile(r'^(?P<key>host_perfdata_file)\s*=\s*(?P<value>.+)$'),
+        re.compile(r'^(?P<key>service_perfdata_file)\s*=\s*(?P<value>.+)$'),
+    ]
+
+    def __init__(self, name, init_config, agentConfig, instances=None):
+        AgentCheck.__init__(self, name, init_config, agentConfig, instances)
+        self.log.setLevel(logging.DEBUG)
+        self.account_id = None
+        self.nagios_tails = {}
+        check_freq = init_config.get("check_freq", 15)
+        if instances is not None:
+            for instance in instances:
+                tailers = []
+                nagios_conf = {}
+                instance_key = None
+
+                if 'nagios_conf' in instance:  # conf.d check
+                    conf_path = instance['nagios_conf']
+                    nagios_conf = self.parse_nagios_config(conf_path)
+                    instance_key = conf_path
+                # Retrocompatibility Code
+                elif 'nagios_perf_cfg' in instance:
```

Comment:
this is not mentioned in the `conf.example.yaml`, would be good to describe it

Reply:
I searched in Nagios docs but could not find this `nagios_perf_cfg` setting. 
I found it only in DataDog Nagios check and here https://github.com/openstack/monasca-agent/blob/master/packaging/monasca-agent/win32/install_files/datadog_win32.conf

I would guess it's from Nagios 3 configuration.
We didn't test check with that Nagios version. Its last release was in 2013.
https://www.nagios.org/projects/nagios-core/history/3x/

Should we remove that Retrocompatibility Code?

Reply:
lets keep notes about supporting old version

Reply:
I added comment with retro compatibility options to `conf.yaml.example`